packages/plugins/RSS3/src/SNSAdaptor/components/FeedCard/CollectibleCard.tsx

```diff
@@ -0,0 +1,258 @@
+import { CardFrame, FeedCardProps } from '../base'
+import { formatValue, Label } from './common'
+
+const useStyles = makeStyles<void, 'image' | 'verbose' | 'info'>()((theme, _, refs) => ({
+    summary: {
+        fontSize: '14px',
+        color: theme.palette.maskColor.third,
+    },
+    verbose: {},
+    image: {},
+    body: {
+        display: 'flex',
+        flexDirection: 'row',
+        alignItems: 'flex-start',
+        marginTop: theme.spacing(1.5),
+        [`&.${refs.verbose}`]: {
+            display: 'block',
+            [`.${refs.image}`]: {
+                width: '100%',
+                height: '100%',
+            },
+            [`.${refs.info}`]: {
+                marginLeft: 0,
+                marginTop: theme.spacing(1.5),
+            },
+        },
+        [`.${refs.image}`]: {
+            width: 64,
+            height: 64,
+            borderRadius: 8,
+            overflow: 'hidden',
+            flexShrink: 0,
+        },
+    },
+    info: {
+        display: 'flex',
+        flexDirection: 'column',
+        justifyContent: 'center',
+        fontSize: 14,
+        color: theme.palette.maskColor.main,
+        lineHeight: '18px',
+        marginLeft: theme.spacing(1.5),
+    },
+    title: {
+        fontWeight: 700,
+    },
+    subtitle: {
+        fontWeight: 400,
+        display: '-webkit-box',
+        WebkitBoxOrient: 'vertical',
+        WebkitLineClamp: 2,
+        overflow: 'hidden',
+    },
+    attributes: {
+        display: 'grid',
+        marginTop: theme.spacing(1.5),
+        gridTemplateColumns: 'repeat(3, 1fr)',
+        gridGap: theme.spacing(2, 2),
+    },
+    attribute: {
+        backgroundColor: theme.palette.maskColor.bg,
+        padding: theme.spacing(1.5),
+        borderRadius: 8,
+        display: 'flex',
+        flexDirection: 'column',
+        alignItems: 'space-between',
+    },
+    attributeType: {
+        color: theme.palette.maskColor.second,
+        textTransform: 'uppercase',
+        fontSize: 14,
+        fontWeight: 400,
+    },
+    attributeValue: {
+        color: theme.palette.maskColor.main,
+        fontSize: 14,
+        fontWeight: 700,
+    },
+}))
+
+const { Tag, Type } = RSS3BaseAPI
+export function isCollectibleFeed(feed: RSS3BaseAPI.Web3Feed): feed is RSS3BaseAPI.CollectibleFeed {
+    return feed.tag === Tag.Collectible
+}
+
+function isRegisteringENS(feed: RSS3BaseAPI.CollectibleFeed) {
+    return feed.actions[1]?.platform === 'ENS Registrar'
+}
+
+interface CollectibleCardProps extends Omit<FeedCardProps, 'feed'> {
+    feed: RSS3BaseAPI.CollectibleFeed
+}
+
+/**
+ * CollectibleCard
+ * Including:
+ *
+ * - CollectibleIn
+ * - CollectibleBurn
+ * - CollectibleIn
+ * - CollectibleMint
+ * - CollectibleOut
+ */
+export const CollectibleCard: FC<CollectibleCardProps> = ({ feed, verbose, ...rest }) => {
+    const { classes, cx } = useStyles()
+
+    const user = useAddressLabel(feed.owner)
+
+    const { metadata, summary } = useMemo(() => {
```

Comment:
Better if refactor to table-driven?

Reply:
There are branches to be checked, for example `ENS`.